Library/Homebrew/os/mac/sdk.rb

```diff
@@ -64,43 +94,31 @@ def sdk_prefix
         ""
       end
 
-      def sdk_paths
-        @sdk_paths ||= begin
-          # Bail out if there is no SDK prefix at all
-          if File.directory? sdk_prefix
-            paths = {}
+      def latest_sdk
+        all_sdks.max_by(&:version)
+      end
 
-            Dir[File.join(sdk_prefix, "MacOSX*.sdk")].each do |sdk_path|
-              version = sdk_path[/MacOSX(\d+\.\d+)u?\.sdk$/, 1]
-              paths[OS::Mac::Version.new(version)] = sdk_path if version.present?
-            end
+      def read_sdk_version(sdk_path)
+        sdk_settings = sdk_path/"SDKSettings.json"
+        sdk_settings_string = sdk_settings.read if sdk_settings.exist?
 
-            # Use unversioned SDK path on Big Sur to avoid issues such as:
-            # https://github.com/Homebrew/homebrew-core/issues/67075
-            # This creates an entry in `paths` whose key is the OS major version
-            sdk_path = Pathname.new("#{sdk_prefix}/MacOSX.sdk")
-            sdk_settings = sdk_path/"SDKSettings.json"
-            if sdk_settings.exist? &&
-               (sdk_settings_string = sdk_settings.read.presence) &&
-               (sdk_settings_json = JSON.parse(sdk_settings_string).presence) &&
-               (version_string = sdk_settings_json.fetch("Version", nil).presence) &&
-               (version = version_string[/(\d+)\./, 1].presence)
-              paths[OS::Mac::Version.new(version)] = sdk_path
-            end
-
-            paths
-          else
-            {}
-          end
+        # Pre-10.14 SDKs
+        sdk_settings = sdk_path/"SDKSettings.plist"
+        if sdk_settings_string.blank? && sdk_settings.exist?
+          result = system_command("plutil", args: ["-convert", "json", "-o", "-", sdk_settings])
+          sdk_settings_string = result.stdout if result.success?
         end
-      end
 
-      # NOTE: This returns a versioned SDK path, even on Big Sur
-      def latest_sdk
-        return if sdk_paths.empty?
+        return if sdk_settings_string.blank?
+        return unless (sdk_settings_json = JSON.parse(sdk_settings_string).presence)
+        return unless (version_string = sdk_settings_json.fetch("Version", nil).presence)
+        return unless (version = Version.new(version_string).major_minor.presence)
 
-        v, path = sdk_paths.max { |(v1, _), (v2, _)| v1 <=> v2 }
-        SDK.new v, path, source
+        if version.major >= 11
```

Comment:
Would be good to make this `11` a constant that's used above too.

Reply:
This is the line I was referring to by "below". https://github.com/Homebrew/brew/pull/10552#discussion_r572008112